x/pairing/keeper/pairing.go

```diff
@@ -286,19 +344,23 @@ func (k Keeper) calculatePairingForClient(ctx sdk.Context, providers []epochstor
 	return validProviders, nil
 }
 
-func (k Keeper) getGeolocationProviders(ctx sdk.Context, providers []epochstoragetypes.StakeEntry, geolocation uint64) []epochstoragetypes.StakeEntry {
+func (k Keeper) getUnfrozenGeolocationProviders(ctx sdk.Context, providers []epochstoragetypes.StakeEntry, geolocation uint64, isSelectedProviders bool) []epochstoragetypes.StakeEntry {
 	validProviders := []epochstoragetypes.StakeEntry{}
 	// create a list of valid providers (stakeAppliedBlock reached)
 	for _, stakeEntry := range providers {
 		if stakeEntry.StakeAppliedBlock > uint64(ctx.BlockHeight()) {
 			// provider stakeAppliedBlock wasn't reached yet
 			continue
 		}
-		geolocationSupported := stakeEntry.Geolocation & geolocation
-		if geolocationSupported == 0 {
-			// no match in geolocation bitmap
-			continue
+
+		if !isSelectedProviders {
```

Comment:
instead of this bool you can set the geolocation to include all (uint max)

Reply:
done